packages/svelte/src/internal/client/dev/log.js

```diff
@@ -0,0 +1,46 @@
+import { STATE_SYMBOL } from '../constants.js';
+import { VERSION } from '../../../version.js';
+import { snapshot } from '../../shared/clone.js';
+import * as w from '../warnings.js';
+
+export function install_custom_formatter() {
+	// Custom formatters are 'supported' in Firefox, but they're worse than useless.
+	// They're not supported in Firefox. We can maybe tweak this over time
+	var is_chrome = typeof navigator !== 'undefined' && navigator.userAgent.includes('Chrome');
+	var custom_formatters_enabled = false;
+
+	if (is_chrome) {
+		// @ts-expect-error
+		(window.devtoolsFormatters ??= []).push({
+			/**
+			 * @param {any} object
+			 * @param {any} config
+			 */
+			header(object, config) {
+				custom_formatters_enabled = true;
+
+				if (STATE_SYMBOL in object) {
+					return [
+						'div',
+						{},
+						['span', { style: 'font-weight: bold' }, '$state'],
+						['object', { object: snapshot(object), config }]
```

Comment:
I think this line is why it's not working in FF. The `"object"` type isn't supported it seems, so we'd have to manually do it with own own styled divs/spans?

Reply:
Ah. Surprising that I learned about `"object"` from firefox-source-docs.mozilla.org of all places. I guess this is why Chrome has 174% market share and Firefox has -42%.

We _could_ construct our own representation but it might be a challenge to make it look consistent with normal object logging across browsers and light/dark modes. We might have to settle for unstyled text if we attempt this.